src/utils/requests/apolloRequests.js

```diff
@@ -12,12 +14,34 @@ const client = new ApolloClient({
   }
 });
 
+export function dispatchError(error) {
+  const { message } = error.graphQLErrors[0];
+  store.dispatch(setError(message));
+  store.dispatch(setFetching(false));
+}
+
 export async function apolloQuery(query) {
-  const result = await client.query({ query });
-  return result.data;
+  store.dispatch(setFetching(true));
+  store.dispatch(setError(''));
+  let result;
+  try {
+    result = await client.query({ query });
+    if (result) store.dispatch(setFetching(false));
+  } catch (error) {
+    dispatchError(error);
+  }
+  return result ? result.data : false;
 }
```

Comment:
Evertime there is a request to mutate/query, it sets state the error and loading automatically. This may not be needed if its just posting but it is reset after each query! Just map loading/error whenever you need to display it to the user

Reply:
Nice! Super efficient.